pkg/streamingpromql/testdata/ours/native_histograms.test

```diff
@@ -147,6 +147,15 @@ eval range from 0 to 4m step 1m sum by (env) (rate(incr_histogram[5m]))
 
 clear
 
+# Test rate() with explicit counter reset information.
+load 1m
+  metric {{sum:3 count:4 buckets:[1 2 1] counter_reset_hint:reset}} {{sum:63 count:124 buckets:[100 4 20] counter_reset_hint:not_reset}}
```

Comment:
can we also write a test that triggers this without a hint?

Reply:
When you say "triggers this", are you referring to the first point being included incorrectly? If so, then I believe the answer is no: the only way `h1.DetectReset(h1)` can return true is if `h1.CounterResetHint` is set to `CounterReset`.